compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/TypeReference.kt

```diff
@@ -146,6 +146,12 @@ public sealed class TypeReference {
       return resolveTypeReference(implementingClass)
     }
 
+    override fun isGenericType(): Boolean {
```

Comment:
So the issue was the wrong implementation for `isGenericType`?

Reply:
Yes, this was returning false for the register use-case, as well as more basic types like Set<String>. The new PSI impl is basically copy-pasta from what we did before the ClassReference refactorings.